docs/hints-for-use-with-docker.md

```diff
@@ -12,7 +12,9 @@ docker run \
   analyze [...] # Insert further arguments for the command.
 ```
 
-**Note:** The single forward slash `/` between the environment variable `$PWD` and the `:` is required for PowerShell compatibility, as PowerShell otherwise interprets `:` as part of the environment variable. 
+**Note:** The single forward slash `/` between the environment variable `$PWD` and the `:` is required for PowerShell compatibility, as PowerShell otherwise interprets `:` as part of the environment variable.
+
+**Note:** When using Yarn/NPM as a projects package manager and mounting a directory relative to the root directory, make sure to **not** have a [repository](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#repository) defined in your `package.json`. As the subdirectory is not recogniced as a VCS managed directory, the repository information cannot be written to the analyzer result. Generating a report will lead to an error.
```

Comment:
I believe this note goes exactly into the wrong direction: Instead of omitting VCS information from `package.json`, you should always mount the root of the project into docker, and eventually adjust the input directory passed to the analyzer to match the subdirectory instead. The point is that ORT needs to be able to "see" the `.git` directory, even if analysis is limited to a subdirectory of the Git working tree.

Reply:
Absolutely. In this case the title is misleading and the "or directory relative to working directory" part must be removed. In this case my note is obsolete.

With "adjust the input directory" you mean using the `-o`flag?

Reply:
No, as this is about the *input* directory, you need to adjust the `-i` option, not the `-o` option.

Let me try to give an example. If on the host you have this directory structure:

```
development
|
+-project-root
  |
  +-.git
  |
  +-sub-project
```

And what you want to get analyzed is only `sub-project`, then you still need to mount `project-root` to Docker, like e.g. `-v /development/project-root:/workdir`, and the analyzer needs to be invoked with `ort analyze -i /workdir/sub-project -o ...`.

Reply:
> No, as this is about the _input_ directory, you need to adjust the `-i` option, not the `-o` option.

Haha, my bad. Sure it's the `-i`option. Small typo, huge effect.

Reply:
Remove note and clarification for working on subprojects